pkg/receive/handler.go

```diff
@@ -625,10 +664,18 @@ func isConflict(err error) bool {
 
 // isNotReady returns whether or not the given error represents a not ready error.
 func isNotReady(err error) bool {
-	return err == tsdb.ErrNotReady ||
+	return err == errNotReady ||
+		err == tsdb.ErrNotReady ||
 		status.Code(err) == codes.Unavailable
 }
 
+// peerState encapsulates the number of request attempt made against a peer and,
+// next allowed time for the next attempt.
+type peerState struct {
```

Comment:
I think we don 't need to give usage context to type name. Something that this allows is rather preferred. For example: `retryState`

and the variable that holds map of it can be named `peerStates` (: